clevertap-core/src/test/java/com/clevertap/android/sdk/LocalDataStoreTest.kt

```diff
@@ -130,26 +132,31 @@ class LocalDataStoreTest : BaseTestCase() {
     }
 
     @Test
-    fun test_getProfileProperty_when_FunctionIsCalledWithSomeKey_should_CallGetProfileValueForKey() {
-        localDataStoreWithConfigSpy.getProfileProperty("key")
-        Mockito.verify(localDataStoreWithConfigSpy, Mockito.times(1)).getProfileValueForKey("key")
+    fun test_getProfileProperty_when_FunctionIsCalledWithSomeKey_should_ReturnAssociatedValue() {
+        localDataStoreWithConfig.updateProfileFields(mapOf("key" to "val"))
+        assertEquals("val", localDataStoreWithConfig.getProfileProperty("key"))
     }
 
     @Test
-    fun test_getProfileValueForKey_when_FunctionIsCalledWithSomeKey_should_ReturnAssociatedValue() {
-        // since getProfileValueForKey() calls _getProfileProperty() which is a private function,
-        // we can't further test it or verify its calling. we can only verify the working of _getProfileProperty
+    fun test_getProfileProperty_when_FunctionIsCalledWithNullKey_should_ReturnNull() {
         localDataStoreWithConfig.getProfileProperty(null).let {
-            println(it)
             assertNull(it)
         }
-        localDataStoreWithConfig.setProfileField("key", "val")
-        localDataStoreWithConfig.getProfileProperty("key").let {
-            println(it)
-            assertNotNull(it)
-            assertEquals("val", it)
-        }
+    }
+
+    @Test
+    fun test_getProfileProperty_when_FunctionIsCalledWithIncorrectKey_should_ReturnNull() {
+        localDataStoreWithConfig.updateProfileFields(mapOf("key" to "val"))
+
+        assertNull(localDataStoreWithConfig.getProfileProperty("key1"))
+    }
+
 
+    @Test
+    fun test_getProfileProperty_when_FunctionIsCalledWithCorrectKeyAndEncryptedValue_should_ReturnNull() {
+        localDataStoreWithConfig.updateProfileFields(mapOf("key" to "[abcd]"))
+
+        assertNull(localDataStoreWithConfig.getProfileProperty("key"))
```

Comment:
assert the value as well, 
localDataStoreWithConfig.getProfileProperty("key") == "[abcd]"

Reply:
getProfileProperty will return null as the value is encrypted